### PR TITLE
Login

### PR DESCRIPTION
**Pull request by plazmap**
Issue line 34 of index.php 

'' login is unspecified index ''

Moi pas comprendre de pourquoi d'abord.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -21,6 +21,29 @@
...
+$app->get('/login', function ($request, $response, $args) {
+    return $this->renderer->render($response, '/login.php', $args);
+});
+
+$app->post('/login', function ($request, $response, $args) {
+    $body = $request->getParsedBody();
+
+    $userinfo=[];
+    $stmt = $this->database->prepare('select name, password from user where name = ?');
+    $stmt->execute($body['login']);
```
Comment (elwinar) [inline index.php]: This should be an array:
```
$stmt->execute([ $body['login'] ]);
```

File: index.php [hunk 2]
```diff
@@ -21,6 +21,28 @@
...
+    return $this->renderer->render($response, '/login.php', $args);
+});
+
+$app->post('/login', function ($request, $response, $args) {
+    $body = $request->getParsedBody();
+
+    $userinfo=[];
+    $stmt = $this->database->prepare('select name, password from user where name = ?');
+    $stmt->execute(array($body['login']));
+    while ($row = $stmt->fetch()){
```
Comment (elwinar) [inline index.php]: There is only one line in the result set, so the loop here isn't useful.

File: index.php [hunk 3]
```diff
@@ -21,6 +21,28 @@
...
+        $userinfo = $row ;
```
Comment (elwinar) [inline index.php]: Here you are not adding the row to the `$userinfo` array, you are replacing the array.

File: index.php [hunk 4]
```diff
@@ -21,6 +21,28 @@
...
+    }
+    foreach ($userinfo as $useri){
+        if ($userinfo['name']==$body['login'] && password_verify($body['login'],$userinfo['password'])){
```
Comment (elwinar) [inline index.php]: The name check isn't useful, as the SQL query ensure that you only have the user that match the form value.

**Review by elwinar — CHANGES_REQUESTED** [review]

File: index.php (hunk 3 above)
Comment (plazmap) [inline index.php]: $userinfo [] = $row ? better ?

Comment (elwinar) [reply index.php]: Better. But still not necessary if you remove the loop altogether.

File: index.php [hunk 5]
```diff
@@ -21,6 +21,26 @@
...
+    return $this->renderer->render($response, '/login.php', $args);
+});
+
+$app->post('/login', function ($request, $response, $args) {
+    $body = $request->getParsedBody();
+
+    $userinfo=[];
+    $stmt = $this->database->prepare('select name, password from user where name = ?');
+    $stmt->execute(array($body['login']));
+    $userinfo = $stmt->fetch();
```
Comment (elwinar) [inline index.php]: `$user` c'est suffisant. Si tu te mets à ajouter des `info` à tout t'es pas arrivé, vu que tout est une information.

File: index.php [hunk 6]
```diff
@@ -21,6 +21,26 @@
...
+ 
+    if (password_verify($body['password'],$userinfo['password'])){
+            echo 'yep';
+        }else{
```
Comment (elwinar) [inline index.php]: Indentation.

**Review by elwinar — CHANGES_REQUESTED** [review]

**Review by elwinar — APPROVED** [review]

**Pull request merged**